docroot/sites/oniowa.uiowa.edu/modules/oniowa_core/oniowa_core.module

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * @file
+ * Primary module hooks for Oniowa Core module.
+ */
+
+use Drupal\Core\Cache\CacheBackendInterface;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Implements hook_form_alter().
+ */
+function oniowa_core_form_alter(&$form, FormStateInterface $form_state, $form_id) {
+  // Use the form['#id'] over $form_id because it is more specific.
+  switch ($form['#id']) {
+    case 'views-exposed-form-events-page-upcoming':
+      $cid = 'sitenow_events:event:dates';
+      $data = \Drupal::cache()->get($cid);
+      if (!$data) {
+        $options = [];
+        $options[''] = t('Select a date');
+        $query = \Drupal::entityQuery('node');
+        $query->condition('type', 'event')
+          ->condition('status', 1)
+          ->condition('field_event_when.end_value', time(), '>')
+          ->sort('field_event_when', 'ASC');
+        $result = $query->execute();
```

Comment:
I think this query can be re-written to return the actual field value we are looking for so that we don't have to load all the nodes below.

Reply:
Is there a way to use `entityQuery` and get the field values directly? There used to be `entityFieldQuery`, but I couldn't find an equivalent. For now, I've rewritten it using `database()` and just querying the field table directly.

Reply:
I was just trying to look that up. I think I might have been getting confused. The documentation for `entityQuery` is pretty light, though, so hard to know for sure. I think what you have looks good. Nice use of `tableExists` for some defensive checking!